app/authentication/token.py

```diff
@@ -45,6 +47,25 @@ async def is_admin(token: str = Depends(oauth2_scheme)) -> bool:
     else:
         return True
 
+# Check is the authorized user is an admin. Return true if so, false if not (with no
+# exception).
+async def is_admin_no_exception(token: str = Depends(oauth2_scheme)) -> bool:
```

Comment:
given how similar this is to `is_admin` above, should we just add `no_exception=bool` keyword argument to `is_admin`

Reply:
Yes, I separated out a function is_app_admin() that handles case checking for gfw or gfwpro admin.  I didn't need the no_exception part (that was really for an earlier version), since I really do want an exception, as long as I can customize the message.